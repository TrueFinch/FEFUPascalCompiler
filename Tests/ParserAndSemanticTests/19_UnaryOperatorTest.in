program a;
var 
    a : integer;
begin
    a := -1;
    writeln(a);
end.