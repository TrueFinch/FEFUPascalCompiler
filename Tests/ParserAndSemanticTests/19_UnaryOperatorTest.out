                a                  
                ┼                  
            MainBlock              
       ┌────────┴───────┐          
      var       CompoundStatement  
       ┼         ┌──────┴┐         
 SimpleVarDecl   :=   writeln      
 ┌─────┼        ┌┴─┐  ┌──┴         
 a  integer     a  -  a            
                   ┼               
                   1               
